src/ucx_van.h

```diff
@@ -591,9 +592,10 @@ class UCXVan : public Van {
       }
 
       do {
-        // Match only 32 bits of tag. Higher bits of tag carry sender id
-        msg = ucp_tag_probe_nb(worker_, UCX_TAG_META,
-                               std::numeric_limits<uint32_t>::max(), 1, &info);
+        /* Only match the highest bit of tag. */
+        ucp_tag_t tag = 0;
```

Comment:
please use `UCX_TAG_META` instead of just number. I think you could redefine `UCX_TAG_META` and `UCX_TAG_DATA` to be last bit of `uint64_t`